slick/src/main/scala/slick/lifted/ExtensionMethods.scala

```diff
@@ -232,6 +232,8 @@ trait ExtensionMethodConversions {
   implicit def numericOptionColumnExtensionMethods[B1](c: Rep[Option[B1]])(implicit tm: BaseTypedType[B1] with NumericTypedType): OptionNumericColumnExtensionMethods[B1] = new OptionNumericColumnExtensionMethods[B1](c)
   implicit def stringColumnExtensionMethods(c: Rep[String]): StringColumnExtensionMethods[String] = new StringColumnExtensionMethods[String](c)
   implicit def stringOptionColumnExtensionMethods(c: Rep[Option[String]]): StringColumnExtensionMethods[Option[String]] = new StringColumnExtensionMethods[Option[String]](c)
+  implicit def mappedToStringColumnExtensionMethods[B1 <: MappedTo[String]](c: Rep[B1])(implicit tm: BaseTypedType[B1]): StringColumnExtensionMethods[String] = new StringColumnExtensionMethods[String](c.asInstanceOf[Rep[String]])
```

Comment:
```
implicit def mappedToStringColumnExtensionMethods[B1 <: MappedTo[String]](c: Rep[B1])(implicit tm: BaseTypedType[B1]): StringColumnExtensionMethods[String] = new StringColumnExtensionMethods[String](c.asInstanceOf[Rep[String]])
```

Why does this require the `tm` implicit parameter?

Reply:
Also, just out of curiosity, what would happen if you returned a `StringColumnExtensionMethods[B1]`?

Reply:
Hmm, it seems like it could be remnants of an other approach. I'll test it out later today, and amend the changes if any are necessary.

Reply:
If you return `B1` you will get compile errors, that about `diverging implicit expansion`, and about `cannot perform option-mapped operation with type`. The implicit parameter is not used, so I've removed it now.